apps/dotcms-ui/src/app/api/services/dot-paginator-es-content/dot-paginator-es-content.service.ts

```diff
@@ -0,0 +1,117 @@
+import { take, map } from 'rxjs/operators';
+import { CoreWebService, ResponseView } from '@dotcms/dotcms-js';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { DotCMSContentlet } from '@dotcms/dotcms-models';
+
+export interface queryEsParams {
+    itemsPerPage?: number;
+    filter?: string;
+    lang?: string;
+    offset?: number;
+    query: string;
+    sortField?: string;
+    sortOrder?: string;
+}
+
+/**
+ * Provides util listing methods to get contentlets data from Elastic Search endpoint
+ * @export
+ * @class DotPaginatorESContentService
+ */
+@Injectable()
+export class DotPaginatorESContentService {
+    paginationPerPage = 40;
+    totalRecords: number;
+
+    private _currentPage: number = 1;
+    private _url = '/api/content/_search';
+    private _defaultQueryParams = { '+languageId': '1', '+deleted': 'false', '+working': 'true' };
+    private _sortField: string = 'modDate';
+    private _sortOrder: string = 'ASC';
+    private _extraParams: Map<string, string> = new Map(Object.entries(this._defaultQueryParams));
+
+    constructor(private coreWebService: CoreWebService) {}
+
+    public get(params: queryEsParams): Observable<DotCMSContentlet[]> {
+        this.setBaseParams(params);
+        this._currentPage = params.offset ? this.getPageFromOffset(params.offset) : 1;
+        const queryParams = this.getESQuery(this.getObjectFromMap(this._extraParams));
+
+        return this.coreWebService
+            .requestView({
+                body: JSON.stringify(queryParams),
+                method: 'POST',
+                url: this._url
+            })
+            .pipe(
+                map((response: ResponseView<any>) => {
+                    this.totalRecords = response.entity?.resultsSize;
```

Comment:
Why this is not part of the response? Having floating properties that we need to update and consume from the component make it harder to debug, use and track.

We should try to mimic the ES response that have all the pagination information in the body.